### DISCUSSION TITLE
Error installing Flux v2 using `eksctl create cluster --config-file`

### DISCUSSION THREAD

**Discussion by jfacevedo** [fluxcd/flux2#4096] [Q&A]
Hi all, I am getting the following error when deploying a Kubernetes cluster with Flux v2:

```
2023-07-22 15:14:01 [ℹ]  gitops configuration detected, setting installer to Flux v2
2023-07-22 15:14:01 [ℹ]  ensuring v1 repo components not installed
2023-07-22 15:14:01 [ℹ]  running pre-flight checks
► checking prerequisites
Assume Role MFA token code: could not get token: EOF
✗ Kubernetes API call failed: Get "https://5A859C6D442D2404A238FE3125FD2398.gr7.us-east-1.eks.amazonaws.com/version": getting credentials: exec: executable aws-iam-authenticator failed with exit code 1
Error: running Flux pre-flight checks: exit status 1
```

The command I am using to deploy it is:

`❯ eksctl create cluster --config-file eks-cluster.yaml --profile integration`

The block I am using to install Flux v2 in the `eks-cluster.yaml` file is:

```
gitops:
  flux:
    gitProvider: github
    flags:
      owner: "mycompany"
      repository: "appv1"
      private: "true"
      branch: "dev"
      namespace: "flux-system"
      read-write-key: "true"
      path: "clusters/dev"
```

**NOTE:** If I remove the Flux block from the `eks-cluster.yaml` file, the cluster deploys without issue. And if I then run the following command to install Flux v2, it installs without issue:

`❯ flux bootstrap github --owner=mycompany --repository=appv1 --branch=dev --path=clusters/dev`

My question is, why does it fail with the gitops block that is inside the `eks-cluster.yaml` file?

I appreciate any help in this regard.

Thanks,
Julian

**Comment by makkes** [comment]
As the issue you're running into is more with eksctl than Flux itself I would like to point you [in their direction](https://github.com/eksctl-io/eksctl#get-in-touch) where you will likely get more help.
